storage/reads/array_cursor.gen.go.tmpl

```diff
@@ -188,6 +210,103 @@ func (c *{{.name}}ArrayCursor) nextArrayCursor() bool {
 	return ok
 }
 
+type {{.name}}FirstArrayCursor struct {
+	cursors.{{.Name}}ArrayCursor
+	res {{$arrayType}}
+	done bool
+}
+
+func new{{.Name}}FirstArrayCursor(cur cursors.{{.Name}}ArrayCursor) *{{.name}}FirstArrayCursor {
+	return &{{.name}}FirstArrayCursor{
+		{{.Name}}ArrayCursor: cur,
+		res: cursors.New{{.Name}}ArrayLen(1),
+	}
+}
+
+func (c *{{.name}}FirstArrayCursor) Stats() cursors.CursorStats { return c.{{.Name}}ArrayCursor.Stats() }
+
+func (c *{{.name}}FirstArrayCursor) Next() {{$arrayType}} {
+	if c.done {
+		return &cursors.{{.Name}}Array{}
+	}
+	a := c.{{.Name}}ArrayCursor.Next()
+	if len(a.Timestamps) == 0 {
+		return a
+	}
+	c.done = true
+	c.res.Timestamps[0] = a.Timestamps[0]
+	c.res.Values[0] = a.Values[0]
+	return c.res
+}
+
+type {{.name}}WindowFirstArrayCursor struct {
+	cursors.{{.Name}}ArrayCursor
+	every int64
+	final int64
+	res {{$arrayType}}
+	tmp {{$arrayType}}
+}
+
+func new{{.Name}}WindowFirstArrayCursor(cur cursors.{{.Name}}ArrayCursor, every int64) *{{.name}}WindowFirstArrayCursor {
+	if every == 0 {
+		every = math.MaxInt64
```

Comment:
I opted for using the zero value to represent "one big window" instead of `math.MaxInt64` for the other windowed aggregates. You might consider following that example here, for consistency's sake. @yzhang1991 and I had a long discussion on it:
https://github.com/influxdata/influxdb/pull/18354#discussion_r435667209
https://github.com/influxdata/influxdb/pull/18414#discussion_r437653943

Reply:
Thanks, will update.